### PR TITLE
Move preaccept expiration logic away from Agent

### PR DESCRIPTION
**Pull request by iamaleksey**

### CODE REVIEW

**Review by belliottsmith — APPROVED** [review]

File: accord-core/src/main/java/accord/coordinate/CoordinateTransaction.java [hunk 1]
```diff
@@ -84,7 +86,7 @@ void onPreAccepted(Topologies topologies, Timestamp executeAt, List<PreAcceptOk>
             //                                  but by sending accept we rule out hybrid fast-path
             // TODO (low priority, efficiency): if we receive an expired response, perhaps defer to permit at least one other
             //                                  node to respond before invalidating
-            if (executeAt.isRejected() || node.agent().isExpired(txnId, executeAt.hlc()))
+            if (executeAt.isRejected() || executeAt.hlc() - txnId.hlc() >= node.agent().preAcceptTimeout(MICROSECONDS))
```
Comment (dcapwell) [inline accord-core/src/main/java/accord/coordinate/CoordinateTransaction.java]: this is true for C* but is this true for accord?  I thought `hlc` was kinda impl specific...

Comment (dcapwell) [reply accord-core/src/main/java/accord/coordinate/CoordinateTransaction.java]: `accord.primitives.Timestamp#hlc`

```
    /**
     * A hybrid logical clock with implementation-defined resolution
     */
    public long hlc()
```

**Review by dcapwell — CHANGES_REQUESTED** [review]
We currently define `hlc` resolution as being implementation specific and this patch assumes microsecond... this works for Cassandra but I don't think this works for accord as it breaks this contract...

File: accord-core/src/main/java/accord/coordinate/CoordinateTransaction.java (hunk 1 above)
Comment (belliottsmith) [reply accord-core/src/main/java/accord/coordinate/CoordinateTransaction.java]: Good catch

Comment (iamaleksey) [reply accord-core/src/main/java/accord/coordinate/CoordinateTransaction.java]: Good point indeed. And it's actually trivially fixed and looks a bit simpler once addressed. Pushed and update.

**Review by ifesdjeen — APPROVED** [review]

**Comment by ifesdjeen** [comment]
My approach to this was slightly different: I have exposed the time resolution via `Node`, but maybe we can just do this later if we see fit.

**Pull request merged**